xgcm/grid.py

```diff
@@ -1150,12 +1150,38 @@ def __init__(
                         f"Input `{dim}` (for the `{pos}` position on axis `{axis}`) is not a dimension in the input datasets `ds`."
                     )
 
+        # Convert all inputs to axes-kwarg mappings
+        # Parse axis properties
+        boundary = self._as_axis_kwarg_mapping(boundary, axes=all_axes)
+        fill_value = self._as_axis_kwarg_mapping(fill_value, axes=all_axes)
+
+        # this case does only apply to periodic. Since we plan on deprecating it soon
+        # handle it here
+        if isinstance(periodic, list):
+            periodic = {axname: True for axname in periodic}
+        periodic = self._as_axis_kwarg_mapping(periodic, axes=all_axes)
+
+        # Set properties on
+
+        # Populate axes. Much of this is just for backward compatibility. I think we are better of
+        # getting completely rid of the axis object and storing/getting info from the grid object properties
+        # (all of which are axis-value mapping).
```

Comment:
This is where type hinting the attributes of the `Grid` class would be helpful

Reply:
I am not quite sure I follow. I see the advantage of doing that, but what exactly is ambiguous here?

Reply:
If you had type hints for the grid object's class attributes then the comment `# (all of which are axis-value mapping).` would be superfluous, because that info would be specified by the type hint.